tinyagentos/notifications.py

```
@@ -130,6 +130,14 @@ async def _post_init(self) -> None:
             await self._db.commit()
         # Create the per-user index if it doesn't exist yet. Cannot live in
         # the SCHEMA because the column is added via guarded ALTER after init.
+        #
+        # NOTE: On first boot after deployment this builds synchronously
+        # inside _post_init (app startup).  On an instance with a large
+        # ``notifications`` table the build takes a write lock on the table
+        # and blocks notification writes for the duration.  This is a
+        # one‑time cost — subsequent starts hit IF NOT EXISTS and return
```

Ruff (0.16.0)

[warning] Comment contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)